main.tf

```diff
@@ -14,38 +14,29 @@ resource "aws_cloudfront_origin_access_identity" "default" {
 
 data "aws_iam_policy_document" "origin" {
   statement {
-    actions   = [ "s3:GetObject" ]
-    resources = [ "arn:aws:s3:::$${bucket_name}$${origin_path}*" ]
+    actions   = ["s3:GetObject"]
+    resources = ["arn:aws:s3:::${local.bucket}${coalesce(var.origin_path, "/")}*"]
 
     principals {
       type        = "AWS"
-      identifiers = [ aws_cloudfront_origin_access_identity.default.iam_arn ]
+      identifiers = [aws_cloudfront_origin_access_identity.default.iam_arn]
     }
   }
 
   statement {
-    actions   = [ "s3:ListBucket" ]
-    resources = [ "arn:aws:s3:::$${bucket_name}" ]
+    actions   = ["s3:ListBucket"]
+    resources = ["arn:aws:s3:::${local.bucket}${var.origin_path}"]
```

Comment:
Is the extra `${var.origin_path}` added on purpose for least priv?
It's hard to follow without seeing a tf plan, but if bucket_name == local.bucket then it seems the origin path part is new.